java/src/main/java/vw/VW.java

```diff
@@ -54,7 +47,44 @@ public static void main(String[] args) {
     public VW(String command) {
         isOpen = true;
         lock = new ReentrantLock();
-        nativePointer = initialize(command);
+        // This allows the user to instead of using the prepackaged JNI layer load their own external JNI layer.
+        // This means that if a user wants to use this code with an OS that is not supported they would follow the following steps
+        // 1.  Download VW
+        // 2.  Build VW for the OS they wish to support
+        // 3.  Call either System.load(xxx) or System.loadLibrary(xxx)
+        // After doing this then the first UnsatisfiedLinkError will never occur.  If a user wishes to use the prepackaed
+        // JNI layer then the first time this object is constructed the UnsatisfiedLinkError will throw and the native libraries
+        // will be loaded and every other time it will not.
+        long currentNativePointer;
+        try {
+            currentNativePointer = initialize(command);
```

Comment:
Based on the above comment, I imagine it's possible to load the native library via: 

```java
System.loadLibrary(...)
```

but the `loadedNativeLibrary` function will still be false.  Is the meaning of the `loadedNativeLibrary` variable really *my load native library procedure was successfully invoked*? Or is should it really mean *someone successfully loaded a valid native VW library*?  If it's the second, I would lock with `STATIC_LOCK` then set `loadedNativeLibrary` to true.  Otherwise, maybe it would be beneficial to have a second variable stating a user-defined native VW library was loaded.